sympy/matrices/matrices.py

```diff
@@ -4260,6 +4260,101 @@ def QRsolve(self, b):
             x.append(tmp / R[j, j])
         return self._new([row._mat for row in reversed(x)])
 
+    def rank_decomposition(self, iszerofunc=_iszero, simplify=False):
+        r"""Returns a pair of matrices (`C`, `F`) with matching rank
+        such that `A = C \cdot F`.
+
+        Parameters
+        ==========
+
+        iszerofunc : Function, optional
+            A function used for detecting whether an element can
+            act as a pivot.  ``lambda x: x.is_zero`` is used by default.
+
+        simplify : Bool or Function, optional
+            A function used to simplify elements when looking for a
+            pivot. By default SymPy's ``simplify`` is used.
+
+        Returns
+        =======
+
+        (C, F) : Matrices
+            `F` is the reduced row echelon form (RREF) of `A`, and `C`
+            has the same rank as `F` and its product with `C` restores `A`:
+            `A = C \cdot F`.
+
+            See Notes for additional mathematical details.
+
+        Examples
+        ========
+
+        >>> from sympy.matrices import Matrix
+        >>> A = Matrix([
+        ...     [1, 3, 1, 4],
+        ...     [2, 7, 3, 9],
+        ...     [1, 5, 3, 1],
+        ...     [1, 2, 0, 8]
+        ... ])
+        >>> C, F = A.rank_decomposition()
+        >>> C
+        Matrix([
+        [1, 3, 4],
+        [2, 7, 9],
+        [1, 5, 1],
+        [1, 2, 8]])
+        >>> F
+        Matrix([
+        [1, 0, -2, 0],
+        [0, 1,  1, 0],
+        [0, 0,  0, 1]])
+        >>> C * F == A
+        True
+
+        Notes
+        =====
+
+        Obtaining `F`, an RREF of `A`, is equivalent to creating a
+        product
+
+        .. math::
+            E_n \cdot E_{n-1} \cdot ... \cdot E_1 \cdot A = F
```

Comment:
I don't like `\cdot` for matrix products...(same in the doc string). Adjacency will do fine.

Reply:
This may need some vote

Reply:
Though I see SymPy's matmul uses adjacency in latex, for now.